sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/SqlDelightPlugin.kt

```diff
@@ -53,14 +53,33 @@ class SqlDelightPlugin : Plugin<Project> {
       val taskName = "generate${it.name.capitalize()}SqlDelightInterface"
       val task = project.tasks.create(taskName, SqlDelightTask::class.java)
       task.group = "sqldelight"
-      task.buildDirectory = File(project.buildDir, "generated")
+      task.buildDirectory = File(project.buildDir, "generated/${it.name}")
       task.description = "Generate Android interfaces for working with ${it.name} database tables"
-      task.source("src")
+      task.source(it.sourceSets.map { "src/${it.name}/${SqlDelightFileType.FOLDER_NAME}" })
       task.include("**${File.separatorChar}*.${SqlDelightFileType.defaultExtension}")
-
-      generateSqlDelight.dependsOn(task)
+      task.packageName = it.packageName()
+      task.sourceFolders = it.sourceSets.map { File("${project.projectDir}/src/${it.name}/${SqlDelightFileType.FOLDER_NAME}") }
 
       it.registerJavaGeneratingTask(task, task.outputDirectory)
     }
   }
+
+  /**
+   * Theres no external api to get the package name. There is to get the application id, but thats
+   * the post build package for the play store, and not the package name that should be used during
+   * compilation. Think R.java, we want to be using the same namespace as it.
+   *
+   * There IS an internal api for doing this.
+   * [BaseVariantImpl.getVariantData().getVariantConfiguration().getPackageFromManifest()],
+   * and so this code just emulates that behavior.
+   *
+   * Package name is enforced identical by agp across multiple source sets, so taking the first
+   * package name we find is fine.
+   */
+  private fun BaseVariant.packageName(): String {
```

Comment:
In AGP 3.2 there will be a stable public API for this. Butter Knife has the same problem...